lib/secure-boot-lib.robot

```diff
@@ -116,19 +116,35 @@ Enable Secure Boot
     [Documentation]    Expects to be executed when in Secure Boot configuration menu.
     [Arguments]    ${sb_menu}
     ${sb_menu}=    Make Sure That Keys Are Provisioned    ${sb_menu}
-    Set Option State    ${sb_menu}    Enable Secure Boot    ${TRUE}
+    ${changed}=    Set Option State    ${sb_menu}    Enable Secure Boot    ${TRUE}
+    IF    ${changed} == ${TRUE}
+        # Changing Secure Boot state issues a special popup
+        Read From Terminal Until    Configuration changed, please reset the platform to take effect!
+        # Dismiss the popup with any key
+        Press Enter
+    END
 
 Disable Secure Boot
     [Documentation]    Expects to be executed when in Secure Boot configuration menu.
     [Arguments]    ${sb_menu}
     ${sb_menu}=    Make Sure That Keys Are Provisioned    ${sb_menu}
-    Set Option State    ${sb_menu}    Enable Secure Boot    ${FALSE}
+    ${changed}=    Set Option State    ${sb_menu}    Enable Secure Boot    ${FALSE}
+    IF    ${changed} == ${TRUE}
+        # Changing Secure Boot state issues a special popup
+        Read From Terminal Until    Configuration changed, please reset the platform to take effect!
+        # Dismiss the popup with any key
+        Press Enter
+    END
 
 Check Secure Boot In Linux
     [Documentation]    Keyword checks Secure Boot state in Linux.
     ...    Returns True when Secure Boot is enabled
     ...    and False when disabled.
-    ${out}=    Execute Linux Command    dmesg | grep secureboot
+    # The string in dmesg may be in two forms:
+    # secureboot: Secure boot disabled
+    # or just:
+    # Secure boot disabled
+    ${out}=    Execute Linux Command    dmesg | grep "Secure boot"
```

Comment:
What are the rules of using `Execute Command In Terminal` vs `Execute Linux Command`?

I can see 235 occurrences of `Execute Linux Command ` still?

Reply:
I linked rules, do you have any specific questions to them? Can you suggest some improvements?

Reply:
"Rules" are clear as is, but they don't seem to be used in practise, hence 235 occurrences of rule-breaking keyword?

Reply:
Yes, because the introduction of quality criteria in the repo was introduced later than the actual code. So we have some state, and set some goals when working on certain parts of the system.

That is why we say: `Whenever you modify a test/keyword, you should rework it to use one of the above. ` - so this applies either new or modified stuff.

This is only one of the examples, it is not really feasible to rework everything at once. Perhaps it would be one of the easiest to replace.

Reply:
https://github.com/Dasharo/open-source-firmware-validation/pull/134/commits/b08c65621929e99a1f41447783e07722f36d9414